File: src/pycropml/transpiler/antlr_py/tests/examples/SimplaceComponent/SoilTemp/src/cs/SoilTemperature/SoilTemperatureExogenous.cs
using System;
using System.Collections.Generic;

public class SoilTemperatureExogenous 
{
    private double _iAirTemperatureMax;
    private double _iAirTemperatureMin;
    private double _iGlobalSolarRadiation;
    private double _iRAIN;
    private double _iCropResidues;
    private double _iPotentialSoilEvaporation;
    private double _iLeafAreaIndex;
    private double[] _SoilTempArray = new double[];
    private double _iSoilWaterContent;
    private double _iSoilSurfaceTemperature;
    
    public SoilTemperatureExogenous() { }
    
    
    public SoilTemperatureExogenous(SoilTemperatureExogenous toCopy, bool copyAll) // copy constructor 
    {
    if (copyAll)
    {
    
    _iAirTemperatureMax = toCopy._iAirTemperatureMax;
    _iAirTemperatureMin = toCopy._iAirTemperatureMin;
    _iGlobalSolarRadiation = toCopy._iGlobalSolarRadiation;
    _iRAIN = toCopy._iRAIN;
    _iCropResidues = toCopy._iCropResidues;
    _iPotentialSoilEvaporation = toCopy._iPotentialSoilEvaporation;
    _iLeafAreaIndex = toCopy._iLeafAreaIndex;
    SoilTempArray = new double[];
            for (int i = 0; i < ; i++)
            { _SoilTempArray[i] = toCopy._SoilTempArray[i]; }
    
    _iSoilWaterContent = toCopy._iSoilWaterContent;
    _iSoilSurfaceTemperature = toCopy._iSoilSurfaceTemperature;
    }
    }
    public double iAirTemperatureMax
        {
            get { return this._iAirTemperatureMax; }
            set { this._iAirTemperatureMax= value; } 
        }
    public double iAirTemperatureMin
        {
            get { return this._iAirTemperatureMin; }
            set { this._iAirTemperatureMin= value; } 
        }
    public double iGlobalSolarRadiation
        {
            get { return this._iGlobalSolarRadiation; }
            set { this._iGlobalSolarRadiation= value; } 
        }
    public double iRAIN
        {
            get { return this._iRAIN; }
            set { this._iRAIN= value; } 
        }
    public double iCropResidues
        {
            get { return this._iCropResidues; }
            set { this._iCropResidues= value; } 
        }
    public double iPotentialSoilEvaporation
        {
            get { return this._iPotentialSoilEvaporation; }
            set { this._iPotentialSoilEvaporation= value; } 
        }
    public double iLeafAreaIndex
        {
            get { return this._iLeafAreaIndex; }
            set { this._iLeafAreaIndex= value; } 
        }
    public double[] SoilTempArray
        {
            get { return this._SoilTempArray; }
            set { this._SoilTempArray= value; } 
        }
    public double iSoilWaterContent
        {
            get { return this._iSoilWaterContent; }
            set { this._iSoilWaterContent= value; } 
        }
    public double iSoilSurfaceTemperature
        {
            get { return this._iSoilSurfaceTemperature; }
            set { this._iSoilSurfaceTemperature= value; } 
        }
}

File: src/pycropml/transpiler/antlr_py/tests/examples/SimplaceComponent/SoilTemp/src/java/SoilTemperature/SoiltemperatureExogenous.java
import  java.io.*;
import  java.util.*;
import java.time.LocalDateTime;

public class SoiltemperatureExogenous
{
    private double iAirTemperatureMax;
    private double iAirTemperatureMin;
    private double iGlobalSolarRadiation;
    private double iRAIN;
    private double iCropResidues;
    private double iPotentialSoilEvaporation;
    private double iLeafAreaIndex;
    private Double [] SoilTempArray;
    private double iSoilWaterContent;
    private double iSoilSurfaceTemperature;
    
    public SoiltemperatureExogenous() { }
    
    public SoiltemperatureExogenous(SoiltemperatureExogenous toCopy, boolean copyAll) // copy constructor 
    {
        if (copyAll)
        {
            this.iAirTemperatureMax = toCopy.getiAirTemperatureMax();
            this.iAirTemperatureMin = toCopy.getiAirTemperatureMin();
            this.iGlobalSolarRadiation = toCopy.getiGlobalSolarRadiation();
            this.iRAIN = toCopy.getiRAIN();
            this.iCropResidues = toCopy.getiCropResidues();
            this.iPotentialSoilEvaporation = toCopy.getiPotentialSoilEvaporation();
            this.iLeafAreaIndex = toCopy.getiLeafAreaIndex();
            SoilTempArray = new Double[toCopy.getSoilTempArray().length];
        for (int i = 0; i < toCopy.getSoilTempArray().length; i++)
        {
            SoilTempArray[i] = toCopy.getSoilTempArray()[i];
        }
            this.iSoilWaterContent = toCopy.getiSoilWaterContent();
            this.iSoilSurfaceTemperature = toCopy.getiSoilSurfaceTemperature();
        }
    }
    public double getiAirTemperatureMax()
    { return iAirTemperatureMax; }

    public void setiAirTemperatureMax(double _iAirTemperatureMax)
    { this.iAirTemperatureMax= _iAirTemperatureMax; } 
    
    public double getiAirTemperatureMin()
    { return iAirTemperatureMin; }

    public void setiAirTemperatureMin(double _iAirTemperatureMin)
    { this.iAirTemperatureMin= _iAirTemperatureMin; } 
    
    public double getiGlobalSolarRadiation()
    { return iGlobalSolarRadiation; }

    public void setiGlobalSolarRadiation(double _iGlobalSolarRadiation)
    { this.iGlobalSolarRadiation= _iGlobalSolarRadiation; } 
    
    public double getiRAIN()
    { return iRAIN; }

    public void setiRAIN(double _iRAIN)
    { this.iRAIN= _iRAIN; } 
    
    public double getiCropResidues()
    { return iCropResidues; }

    public void setiCropResidues(double _iCropResidues)
    { this.iCropResidues= _iCropResidues; } 
    
    public double getiPotentialSoilEvaporation()
    { return iPotentialSoilEvaporation; }

    public void setiPotentialSoilEvaporation(double _iPotentialSoilEvaporation)
    { this.iPotentialSoilEvaporation= _iPotentialSoilEvaporation; } 
    
    public double getiLeafAreaIndex()
    { return iLeafAreaIndex; }

    public void setiLeafAreaIndex(double _iLeafAreaIndex)
    { this.iLeafAreaIndex= _iLeafAreaIndex; } 
    
    public Double [] getSoilTempArray()
    { return SoilTempArray; }

    public void setSoilTempArray(Double [] _SoilTempArray)
    { this.SoilTempArray= _SoilTempArray; } 
    
    public double getiSoilWaterContent()
    { return iSoilWaterContent; }

    public void setiSoilWaterContent(double _iSoilWaterContent)
    { this.iSoilWaterContent= _iSoilWaterContent; } 
    
    public double getiSoilSurfaceTemperature()
    { return iSoilSurfaceTemperature; }

    public void setiSoilSurfaceTemperature(double _iSoilSurfaceTemperature)
    { this.iSoilSurfaceTemperature= _iSoilSurfaceTemperature; } 
    
}